src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -406,16 +408,16 @@ export const useSwapper = () => {
     tokenId?: string
   }): Promise<string> => {
     const swapper = await swapperManager.getBestSwapper({
-      buyAssetId: quote.buyAsset.assetId,
-      sellAssetId: quote.sellAsset.assetId,
+      buyAssetId: buyAsset.currency.assetId,
+      sellAssetId: sellAsset.currency.assetId,
     })
     return swapper?.getUsdRate({
       symbol,
       tokenId,
     })
   }
 
-  const setFees = async (result: Quote<ChainTypes>, sellAsset: Asset) => {
+  const setFees = async (result: Trade<ChainTypes> | TradeQuote<ChainTypes>, sellAsset: Asset) => {
```

Comment:
```suggestion
  const setFees = async (fees: Pick<Trade<ChainTypes> | TradeQuote<ChainTypes>, 'fees'>, sellAsset: Asset) => {
```

Reply:
I couldnt make this work

Reply:
I suspect he meant `feeData` instead of `fees`:

```suggestion
  const setFees = async (fees: Pick<Trade<ChainTypes> | TradeQuote<ChainTypes>, 'feeData'>, sellAsset: Asset) => {
```

Reply:
Derp. That was it, thanks